src/tileray.cpp

```diff
@@ -197,7 +197,7 @@ void tileray::advance( int num )
         return;
     }
     int anum = std::abs( num );
-    steps = anum;
+    steps += anum;
```

Comment:
These tileray changes are fundamental enough that I'd like to see them broken out into a separate commit.

Also, this seems like a substantial change to tilerays.  Obviously, it is correct, but would you please explain this change?  Not in a comment in the code, but just a few words here.

Reply:
When you drive at a slight angle you're mostly going forward, but occasionally move 1 tile sideways. In order to predict exactly when that will happen you need to keep track of how many times the vehicle.face tileray has been advanced since the last change in direction (in order to be able to create an identical tileray in the simulation). This is why navigation_node.tileray_steps exists.

The += change here is a bug fix. It was unnoticed because nobody used the steps value for anything.

I could put the tileray changes into a separate PR, but I'd be adding a function that nobody calls. Is that ok?

Reply:
I don't think it needs to be in a separate PR, necessarily, just in a separate commit in this PR.  More or less a request for when you rebase.

Reply:
Done.